doc/user/flycheck-versus-flymake.rst

```diff
@@ -44,71 +37,68 @@ Please do **not** use this table alone to make your personal judgment.  Read the
++---------------------------+-----------------------+-----------------------+
+|`Enables automatically if  |yes                    |no                     |
+|possible <Enabling Syntax  |                       |                       |
+|Checking_>`_               |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Checks after               |save, newline, change  |newline, change        |
++---------------------------+-----------------------+-----------------------+
+|Checks in background       |yes                    |yes                    |
++---------------------------+-----------------------+-----------------------+
+|`Automatic syntax checker  |by major mode and      |user-defined [#]_      |
+|selection <Syntax checker  |custom predicates      |                       |
+|selection_>`_              |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Manual syntax checker     |yes                    |yes                    |
+|selection <Manual          |                       |                       |
+|Selection_>`_              |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Multiple syntax checkers  |yes                    |yes                    |
+|per buffer`_               |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Supported languages`_     |>50                    |2 [#]_                 |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Checking remote files      |said to work, but not  |partly?                |
+|via Tramp                  |officially supported   |                       |
+|                           |[#]_                   |                       |
++---------------------------+-----------------------+-----------------------+
+|`Definition of new         |single declarative     |function definition    |
+|syntax checkers`_          |function/macro         |[#]_                   |
++---------------------------+-----------------------+-----------------------+
+|`Error levels`_            |errors, warnings,      |errors, warnings,      |
+|                           |infos, custom levels   |notes, custom levels   |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Error identifiers`_       |yes                    |no                     |
++---------------------------+-----------------------+-----------------------+
+|`Error parsing`_           |can use arbitrary      |can use arbitrary      |
```

Comment:
I don't know exactly what built-in-helpers are...

Reply:
In this case, functions that do part of the parsing for you like `flycheck-parse-json` or `flycheck-parse-checkstyle`.  Certainly not anything that Flymake could not provide.

Reply:
OK. No flymake doesn't have build-in helpers for parsing diagnostics. It might try to use `compilation-mode`s in the future, which is a huge database of those.